Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs

```diff
@@ -534,6 +560,46 @@ bool InitFromFile(string path) {
             RefreshRuntimeSettings();
         }
 
+        async void InitializeDeviceData() {
+            bool InitFromFile(string path) {
+                if (DeserializeFromFile(path, out string contents, out DeviceData deviceData)) {
+                    lastDeviceDataJSON = contents;
+                    DeviceData = deviceData;
+                    OnDeviceDataChange?.Invoke(deviceData);
+                    return true;
+                }
+                else
+                    return false;
+            }
+
+            string filePath = _cachedDeviceStatusPath;
+
+            // Method 1: Try to initialize using the cached json file
+            if (LoggingEnabled)
+                Debug.unityLogger.LogWarning(TAG, "Checking if DeviceData can be initialized using the cached json file.");
+
+            if (InitFromFile(filePath)) {
+                if (LoggingEnabled)
+                    Debug.unityLogger.Log(TAG, "Initialized DeviceData using cached json file. ");
+                return;
+            }
+
+            // Method 2: If initialization using cached json fails, we wait for the SDK to get connected
+            // to the admin app and then request a DeviceData refresh
+            string msg = "Cannot initialize DeviceData using cached json file.";
+            if (!IsConnectedToAdminApp)
+                msg += "Waiting for MXR Admin App connection to send it a DeviceData refresh request.";
+            if (LoggingEnabled)
+                Debug.unityLogger.LogWarning(TAG, msg);
+
+            while (!IsConnectedToAdminApp)
+                await Task.Delay(100);
```

Comment:
You mentioned we now *require* the admin app to send a DeviceData object. 

I think this is a bit dangerous to check in, since:

1. Admin app download could fail / we need to be really tight on release timing with HS
2. We have lots of old admin app versions in dev this will cause compatibility issues with. 

Can we fallback gracefully or gate this on a specific version of the Admin App?

Reply:
Oh ok. For compatibility with older versions we can gate with admin app version. 

Not sure if I understand point 1, can sync when you're available

Reply:
To expand on point 1:

1. We may want to release home screen before admin app, and checking this code in with blocking behavior would prevent that. We should always avoid checking in code that blocks the release, preferring to gate or flag
2. If we push admin app and home screen at the same time, the admin app may fail to download itself, but succeed on home screen, causing a borked home screen until admin app installs successfully. 

It's all the same idea really - compatibility with old versions.

Reply:
Got it! Will gate this code behind an admin app version code once we have an admin app build with the DeviceData messaging

Reply:
PR #10 on this stack handles admin app version gating. Resolving this